### PR TITLE
Adds diffLink macro to core. Fix #1539.

### LINKED ISSUE #1539: Add diffLink to core

### PR DESCRIPTION
**Pull request by AndreaLaGrotteria**

### CODE REVIEW

File: buildSrc/src/main/java/io/micronaut/guides/core/BuildDiffLinkSubstitution.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                appName = extractAppName(line);
+            } else {
+                appName = guidesConfiguration.getDefaultAppName();
+            }
+            App app = guide.apps().stream()
+                    .filter(a -> a.name().equals(appName))
+                    .findFirst()
+                    .orElse(null);
+
+            String link = "https://micronaut.io/launch?" +
```
Comment (sdelamo) [inline buildSrc/src/main/java/io/micronaut/guides/core/BuildDiffLinkSubstitution.java]: Please, used the Micronaut `UriBuilder` api.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/java/io/micronaut/guides/core/GuidesConfigurationProperties.java [hunk 2]
```diff
@@ -58,6 +60,9 @@ public String getDefaultAppName() {
         return DEFAULT_APP_NAME;
     }
 
+    @Override
+    public String getLauncherUrl() { return LAUNCHER_URL; }
```
Comment (sdelamo) [inline buildSrc/src/main/java/io/micronaut/guides/core/GuidesConfigurationProperties.java]: rename to `getProjectGeneratorUrl()`

File: buildSrc/src/main/java/io/micronaut/guides/core/BuildDiffLinkSubstitution.java [hunk 3]
```diff
@@ -29,21 +28,20 @@ public String substitute(String str, GuidesOption option, Guide guide) {
...
+                    .queryParam("lang", option.getLanguage().name())
+                    .queryParam("build", option.getBuildTool().name())
+                    .queryParam("test", option.getTestFramework().name())
+                    .queryParam("name", appName.equals(guidesConfiguration.getDefaultAppName()) ? "micronautguide" : appName)
+                    .queryParam("type", app.applicationType().name())
+                    .queryParam("package", guidesConfiguration.getPackageName())
+                    .queryParam("activity", "diff");
+            featureNames(line, app, option).forEach(f -> uriBuilder.queryParam("features", f));
+
+            String res = "NOTE: If you have an existing Micronaut application and want to add the functionality described here, you can "
```
Comment (sdelamo) [inline buildSrc/src/main/java/io/micronaut/guides/core/BuildDiffLinkSubstitution.java]: I think his copy should be in a common snippet instead. I don't think we should have copy trapped in the java classes.

**Review by sdelamo — CHANGES_REQUESTED** [review]

File: buildSrc/src/main/java/io/micronaut/guides/core/MacroSubstitution.java [hunk 4]
```diff
@@ -1,10 +1,33 @@
...
+
+import java.util.List;
 
 public interface MacroSubstitution {
     String APP = "app";
+    String APP_NAME_DEFAULT = "default";
 
     @NonNull
-    String substitute(@NonNull String str, @NonNull String slug, @NonNull GuidesOption option);
+    String substitute(@NonNull String str, @NonNull Guide slug, @NonNull GuidesOption option);
```
Comment (AndreaLaGrotteria) [inline buildSrc/src/main/java/io/micronaut/guides/core/MacroSubstitution.java]: Should probably be `Guide guide`

**Pull request merged**